little hack to make elasticsearch search OK on indexes WITHOUT any "metadata" field

RodolpheCalvet opened the pull request:
**Description:**
This is for enabling the whole thing for people (like me) who just **don't have "metadata" field in their index**... these extra things I put them at the same level as content in the doc, so using your tools was asking me to add a useless metadata field... I prefered to share this.
I Tested it, looks ok! The search returns the expected doc without any metadata part

**Issue:**
Blocking the search if docs DONT have the "metadata" field.

**Dependencies:**
no change

RodolpheCalvet committed:
Update elasticsearch.py

This is for enabling the whole thing for people (like me) who just don't have "metadata" field in their index... these extra things I put them at the same level as content in the doc.
Tested it, looks ok!

RodolpheCalvet committed:
Merge pull request #1 from RodolpheCalvet/RodolpheCalvet-patch-1

Update elasticsearch.py

baskaryan commented:
cc @joemcelroy

efriis commented:
Also if you could get lint passing that would be great! @RodolpheCalvet

RodolpheCalvet committed:
Merge branch 'master' into master

joemcelroy commented on libs/langchain/langchain/vectorstores/elasticsearch.py:
```diff
@@ -786,7 +786,7 @@ def _search(
             return Document(
                 page_content=hit["_source"].get(self.query_field, ""),
-                metadata=hit["_source"]["metadata"],
+                metadata=hit["_source"]["metadata"] if "metadata" in hit["_source"] else {},
```
```suggestion
                metadata=hit["_source"].get("metadata", {}),
```

Could you also add tests for this please?

joemcelroy commented:
hi there @RodolpheCalvet - Thanks for your contribution, this is a good addition :)

Yes you're fine to update the existing PR.

RodolpheCalvet committed:
Update elasticsearch.py

Commit on the main branch in order to avoid a new PR (beginner..;)

RodolpheCalvet committed:
Update elasticsearch.py

Your suggested use of get() method works as the previous suggested did. So i got rid of the boolean lines upwards and put your suggestion.

RodolpheCalvet committed:
Update elasticsearch.py

Due to lint I put back the coma at the end of the line

RodolpheCalvet committed:
Merge branch 'langchain-ai:master' into master

RodolpheCalvet committed:
Merge branch 'langchain-ai:master' into master

RodolpheCalvet committed:
Update elasticsearch.py

My last hack for reaching the mitigation IF clause

RodolpheCalvet committed:
Update pyproject.toml

enabling pip git from my fork cf https://stackoverflow.com/questions/72294299/multiple-top-level-packages-discovered-in-a-flat-layout

RodolpheCalvet committed:
Update pyproject.toml

typo

RodolpheCalvet committed:
Update pyproject.toml

"langchain" plutot que null

RodolpheCalvet committed:
Update pyproject.toml

yo

RodolpheCalvet committed:
Update pyproject.toml

bou

RodolpheCalvet committed:
Update pyproject.toml

RodolpheCalvet committed:
Update pyproject.toml

RodolpheCalvet committed:
Update pyproject.toml

RodolpheCalvet committed:
Update pyproject.toml

RodolpheCalvet committed:
Update pyproject.toml

RodolpheCalvet committed:
Update pyproject.toml

RodolpheCalvet committed:
Update pyproject.toml

baskaryan commented:
elasticsearch integration has been moved to it's own package and repo, we should re-open this fix against langchain-elastic repo here: https://github.com/langchain-ai/langchain-elastic/blob/main/libs/elasticsearch/langchain_elasticsearch/vectorstores.py
